### PR TITLE
Copy edits for data-table usage content

### PR DESCRIPTION
**Pull request by crlordan**
Working on changes in this issue: https://github.com/carbon-design-system/carbon-website/issues/4235

Working to improve the content of the data-table component.

#### Changelog

**Changed**

- Copy edits for the Universal Behaviors section.
- Added a subhead to the entry in the Modifiers section, to match other components and label the modified more clearly.

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Hi, thanks for the contribution, you'll want to run `yarn format` locally for ci to pass

**Comment by crlordan** [comment]
I ran yarn format and now the ci checks have passed.

**Comment by crlordan** [comment]
@alisonjoseph Hi, wondering if this PR could be reviewed and merged now that the checks have passed? Thank you.

File: src/pages/components/data-table/usage.mdx [hunk 1]
```diff
@@ -429,16 +429,16 @@ The expandable icon always appears first and to the left of the selection icon.
...
+in other Carbon components. Buttons, checkboxes, radio buttons, overflow menus,
+search, and expanding elements are used in the data table, and all behave
+according to their guidelines.
 
 #### Mouse
 
 Users can trigger an action or function in the table by clicking the associated
 button or component. Each action or function has its own distinct click target.
-See the accessibility tab for additional interactive guidance.
+See the Accessibility tab for more information.
```
Comment (aagonzales) [inline src/pages/components/data-table/usage.mdx:441]: Should we hyperlink this? And would the link be just to word "Accessibility" or "Accessibility tab"? 
```suggestion
See the [Accessibility](/components/data-table/accessibility/) tab for more information.
```

**Review by aagonzales — APPROVED** [review]
These changes look good to me! I had one suggestion of adding a crosslink to the Accessibility tab.

**Review by alisonjoseph — APPROVED** [review]

**Review by emyarod — COMMENTED** [review]
it looks like the `format` check is still failing?

**Pull request merged**